src/ComposerEventHandler.php

```diff
@@ -85,7 +89,9 @@ public function onPostAutoloadDump(Event $event): void
         $outputDirectory = $this->getPluginOutputDirectory($rootPackage);
         $this->ensureDirectoryExists($outputDirectory);
 
-        $packages = $composer->getRepositoryManager()->getLocalRepository()->getPackages();
+        $packages = count($this->updatedPackages) === 0
```

Comment:
Am I correct that when there are updated packaes merge plan will only include these packages? If so, that sounds wrong since merge plan should be generated for all packages available.

Reply:
Fixed.